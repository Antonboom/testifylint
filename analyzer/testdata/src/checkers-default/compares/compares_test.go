// Code generated by testifylint/internal/testgen. DO NOT EDIT.

package compares

import (
	"testing"

	"github.com/stretchr/testify/assert"
)

func TestComparesChecker(t *testing.T) {
	var a, b int
	var c, d bool
	var ptrA, ptrB *int

	// Invalid.
	{
		assert.True(t, a == b)                                          // want "compares: use assert\\.Equal"
		assert.Truef(t, a == b, "msg with args %d %s", 42, "42")        // want "compares: use assert\\.Equalf"
		assert.True(t, a != b)                                          // want "compares: use assert\\.NotEqual"
		assert.Truef(t, a != b, "msg with args %d %s", 42, "42")        // want "compares: use assert\\.NotEqualf"
		assert.True(t, a > b)                                           // want "compares: use assert\\.Greater"
		assert.Truef(t, a > b, "msg with args %d %s", 42, "42")         // want "compares: use assert\\.Greaterf"
		assert.True(t, a >= b)                                          // want "compares: use assert\\.GreaterOrEqual"
		assert.Truef(t, a >= b, "msg with args %d %s", 42, "42")        // want "compares: use assert\\.GreaterOrEqualf"
		assert.True(t, a < b)                                           // want "compares: use assert\\.Less"
		assert.Truef(t, a < b, "msg with args %d %s", 42, "42")         // want "compares: use assert\\.Lessf"
		assert.True(t, a <= b)                                          // want "compares: use assert\\.LessOrEqual"
		assert.Truef(t, a <= b, "msg with args %d %s", 42, "42")        // want "compares: use assert\\.LessOrEqualf"
		assert.False(t, a == b)                                         // want "compares: use assert\\.NotEqual"
		assert.Falsef(t, a == b, "msg with args %d %s", 42, "42")       // want "compares: use assert\\.NotEqualf"
		assert.False(t, a != b)                                         // want "compares: use assert\\.Equal"
		assert.Falsef(t, a != b, "msg with args %d %s", 42, "42")       // want "compares: use assert\\.Equalf"
		assert.False(t, a > b)                                          // want "compares: use assert\\.LessOrEqual"
		assert.Falsef(t, a > b, "msg with args %d %s", 42, "42")        // want "compares: use assert\\.LessOrEqualf"
		assert.False(t, a >= b)                                         // want "compares: use assert\\.Less"
		assert.Falsef(t, a >= b, "msg with args %d %s", 42, "42")       // want "compares: use assert\\.Lessf"
		assert.False(t, a < b)                                          // want "compares: use assert\\.GreaterOrEqual"
		assert.Falsef(t, a < b, "msg with args %d %s", 42, "42")        // want "compares: use assert\\.GreaterOrEqualf"
		assert.False(t, a <= b)                                         // want "compares: use assert\\.Greater"
		assert.Falsef(t, a <= b, "msg with args %d %s", 42, "42")       // want "compares: use assert\\.Greaterf"
		assert.NotEmpty(t, a == b)                                      // want "compares: use assert\\.Equal"
		assert.NotEmptyf(t, a == b, "msg with args %d %s", 42, "42")    // want "compares: use assert\\.Equalf"
		assert.NotEmpty(t, a != b)                                      // want "compares: use assert\\.NotEqual"
		assert.NotEmptyf(t, a != b, "msg with args %d %s", 42, "42")    // want "compares: use assert\\.NotEqualf"
		assert.NotEmpty(t, a > b)                                       // want "compares: use assert\\.Greater"
		assert.NotEmptyf(t, a > b, "msg with args %d %s", 42, "42")     // want "compares: use assert\\.Greaterf"
		assert.NotEmpty(t, a >= b)                                      // want "compares: use assert\\.GreaterOrEqual"
		assert.NotEmptyf(t, a >= b, "msg with args %d %s", 42, "42")    // want "compares: use assert\\.GreaterOrEqualf"
		assert.NotEmpty(t, a < b)                                       // want "compares: use assert\\.Less"
		assert.NotEmptyf(t, a < b, "msg with args %d %s", 42, "42")     // want "compares: use assert\\.Lessf"
		assert.NotEmpty(t, a <= b)                                      // want "compares: use assert\\.LessOrEqual"
		assert.NotEmptyf(t, a <= b, "msg with args %d %s", 42, "42")    // want "compares: use assert\\.LessOrEqualf"
		assert.Empty(t, a == b)                                         // want "compares: use assert\\.NotEqual"
		assert.Emptyf(t, a == b, "msg with args %d %s", 42, "42")       // want "compares: use assert\\.NotEqualf"
		assert.Empty(t, a != b)                                         // want "compares: use assert\\.Equal"
		assert.Emptyf(t, a != b, "msg with args %d %s", 42, "42")       // want "compares: use assert\\.Equalf"
		assert.Empty(t, a > b)                                          // want "compares: use assert\\.LessOrEqual"
		assert.Emptyf(t, a > b, "msg with args %d %s", 42, "42")        // want "compares: use assert\\.LessOrEqualf"
		assert.Empty(t, a >= b)                                         // want "compares: use assert\\.Less"
		assert.Emptyf(t, a >= b, "msg with args %d %s", 42, "42")       // want "compares: use assert\\.Lessf"
		assert.Empty(t, a < b)                                          // want "compares: use assert\\.GreaterOrEqual"
		assert.Emptyf(t, a < b, "msg with args %d %s", 42, "42")        // want "compares: use assert\\.GreaterOrEqualf"
		assert.Empty(t, a <= b)                                         // want "compares: use assert\\.Greater"
		assert.Emptyf(t, a <= b, "msg with args %d %s", 42, "42")       // want "compares: use assert\\.Greaterf"
		assert.NotZero(t, a == b)                                       // want "compares: use assert\\.Equal"
		assert.NotZerof(t, a == b, "msg with args %d %s", 42, "42")     // want "compares: use assert\\.Equalf"
		assert.NotZero(t, a != b)                                       // want "compares: use assert\\.NotEqual"
		assert.NotZerof(t, a != b, "msg with args %d %s", 42, "42")     // want "compares: use assert\\.NotEqualf"
		assert.NotZero(t, a > b)                                        // want "compares: use assert\\.Greater"
		assert.NotZerof(t, a > b, "msg with args %d %s", 42, "42")      // want "compares: use assert\\.Greaterf"
		assert.NotZero(t, a >= b)                                       // want "compares: use assert\\.GreaterOrEqual"
		assert.NotZerof(t, a >= b, "msg with args %d %s", 42, "42")     // want "compares: use assert\\.GreaterOrEqualf"
		assert.NotZero(t, a < b)                                        // want "compares: use assert\\.Less"
		assert.NotZerof(t, a < b, "msg with args %d %s", 42, "42")      // want "compares: use assert\\.Lessf"
		assert.NotZero(t, a <= b)                                       // want "compares: use assert\\.LessOrEqual"
		assert.NotZerof(t, a <= b, "msg with args %d %s", 42, "42")     // want "compares: use assert\\.LessOrEqualf"
		assert.Zero(t, a == b)                                          // want "compares: use assert\\.NotEqual"
		assert.Zerof(t, a == b, "msg with args %d %s", 42, "42")        // want "compares: use assert\\.NotEqualf"
		assert.Zero(t, a != b)                                          // want "compares: use assert\\.Equal"
		assert.Zerof(t, a != b, "msg with args %d %s", 42, "42")        // want "compares: use assert\\.Equalf"
		assert.Zero(t, a > b)                                           // want "compares: use assert\\.LessOrEqual"
		assert.Zerof(t, a > b, "msg with args %d %s", 42, "42")         // want "compares: use assert\\.LessOrEqualf"
		assert.Zero(t, a >= b)                                          // want "compares: use assert\\.Less"
		assert.Zerof(t, a >= b, "msg with args %d %s", 42, "42")        // want "compares: use assert\\.Lessf"
		assert.Zero(t, a < b)                                           // want "compares: use assert\\.GreaterOrEqual"
		assert.Zerof(t, a < b, "msg with args %d %s", 42, "42")         // want "compares: use assert\\.GreaterOrEqualf"
		assert.Zero(t, a <= b)                                          // want "compares: use assert\\.Greater"
		assert.Zerof(t, a <= b, "msg with args %d %s", 42, "42")        // want "compares: use assert\\.Greaterf"
		assert.True(t, ptrA == ptrB)                                    // want "compares: use assert\\.Same"
		assert.Truef(t, ptrA == ptrB, "msg with args %d %s", 42, "42")  // want "compares: use assert\\.Samef"
		assert.True(t, ptrA != ptrB)                                    // want "compares: use assert\\.NotSame"
		assert.Truef(t, ptrA != ptrB, "msg with args %d %s", 42, "42")  // want "compares: use assert\\.NotSamef"
		assert.False(t, ptrA == ptrB)                                   // want "compares: use assert\\.NotSame"
		assert.Falsef(t, ptrA == ptrB, "msg with args %d %s", 42, "42") // want "compares: use assert\\.NotSamef"
		assert.False(t, ptrA != ptrB)                                   // want "compares: use assert\\.Same"
		assert.Falsef(t, ptrA != ptrB, "msg with args %d %s", 42, "42") // want "compares: use assert\\.Samef"
	}

	// Valid.
	{
		assert.Equal(t, a, b)
		assert.Equalf(t, a, b, "msg with args %d %s", 42, "42")
		assert.NotEqual(t, a, b)
		assert.NotEqualf(t, a, b, "msg with args %d %s", 42, "42")
		assert.Greater(t, a, b)
		assert.Greaterf(t, a, b, "msg with args %d %s", 42, "42")
		assert.GreaterOrEqual(t, a, b)
		assert.GreaterOrEqualf(t, a, b, "msg with args %d %s", 42, "42")
		assert.Less(t, a, b)
		assert.Lessf(t, a, b, "msg with args %d %s", 42, "42")
		assert.LessOrEqual(t, a, b)
		assert.LessOrEqualf(t, a, b, "msg with args %d %s", 42, "42")
		assert.Same(t, ptrA, ptrB)
		assert.Samef(t, ptrA, ptrB, "msg with args %d %s", 42, "42")
		assert.NotSame(t, ptrA, ptrB)
		assert.NotSamef(t, ptrA, ptrB, "msg with args %d %s", 42, "42")
	}

	// Ignored.
	{
		assert.True(t, c && d)
		assert.Truef(t, c && d, "msg with args %d %s", 42, "42")
		assert.False(t, d && c)
		assert.Falsef(t, d && c, "msg with args %d %s", 42, "42")
		assert.True(t, c || d)
		assert.Truef(t, c || d, "msg with args %d %s", 42, "42")
		assert.False(t, d || c)
		assert.Falsef(t, d || c, "msg with args %d %s", 42, "42")
	}
}
