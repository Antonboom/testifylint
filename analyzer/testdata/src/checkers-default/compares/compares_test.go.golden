// Code generated by testifylint/internal/testgen. DO NOT EDIT.

package compares

import (
	"testing"

	"github.com/stretchr/testify/assert"
)

func TestComparesChecker(t *testing.T) {
	var a, b int
	var c, d bool
	var ptrA, ptrB *int

	// Invalid.
	{
		assert.Equal(t, a, b)                                            // want "compares: use assert\\.Equal"
		assert.Equalf(t, a, b, "msg with args %d %s", 42, "42")          // want "compares: use assert\\.Equalf"
		assert.NotEqual(t, a, b)                                         // want "compares: use assert\\.NotEqual"
		assert.NotEqualf(t, a, b, "msg with args %d %s", 42, "42")       // want "compares: use assert\\.NotEqualf"
		assert.Greater(t, a, b)                                          // want "compares: use assert\\.Greater"
		assert.Greaterf(t, a, b, "msg with args %d %s", 42, "42")        // want "compares: use assert\\.Greaterf"
		assert.GreaterOrEqual(t, a, b)                                   // want "compares: use assert\\.GreaterOrEqual"
		assert.GreaterOrEqualf(t, a, b, "msg with args %d %s", 42, "42") // want "compares: use assert\\.GreaterOrEqualf"
		assert.Less(t, a, b)                                             // want "compares: use assert\\.Less"
		assert.Lessf(t, a, b, "msg with args %d %s", 42, "42")           // want "compares: use assert\\.Lessf"
		assert.LessOrEqual(t, a, b)                                      // want "compares: use assert\\.LessOrEqual"
		assert.LessOrEqualf(t, a, b, "msg with args %d %s", 42, "42")    // want "compares: use assert\\.LessOrEqualf"
		assert.NotEqual(t, a, b)                                         // want "compares: use assert\\.NotEqual"
		assert.NotEqualf(t, a, b, "msg with args %d %s", 42, "42")       // want "compares: use assert\\.NotEqualf"
		assert.Equal(t, a, b)                                            // want "compares: use assert\\.Equal"
		assert.Equalf(t, a, b, "msg with args %d %s", 42, "42")          // want "compares: use assert\\.Equalf"
		assert.LessOrEqual(t, a, b)                                      // want "compares: use assert\\.LessOrEqual"
		assert.LessOrEqualf(t, a, b, "msg with args %d %s", 42, "42")    // want "compares: use assert\\.LessOrEqualf"
		assert.Less(t, a, b)                                             // want "compares: use assert\\.Less"
		assert.Lessf(t, a, b, "msg with args %d %s", 42, "42")           // want "compares: use assert\\.Lessf"
		assert.GreaterOrEqual(t, a, b)                                   // want "compares: use assert\\.GreaterOrEqual"
		assert.GreaterOrEqualf(t, a, b, "msg with args %d %s", 42, "42") // want "compares: use assert\\.GreaterOrEqualf"
		assert.Greater(t, a, b)                                          // want "compares: use assert\\.Greater"
		assert.Greaterf(t, a, b, "msg with args %d %s", 42, "42")        // want "compares: use assert\\.Greaterf"
		assert.Equal(t, a, b)                                            // want "compares: use assert\\.Equal"
		assert.Equalf(t, a, b, "msg with args %d %s", 42, "42")          // want "compares: use assert\\.Equalf"
		assert.NotEqual(t, a, b)                                         // want "compares: use assert\\.NotEqual"
		assert.NotEqualf(t, a, b, "msg with args %d %s", 42, "42")       // want "compares: use assert\\.NotEqualf"
		assert.Greater(t, a, b)                                          // want "compares: use assert\\.Greater"
		assert.Greaterf(t, a, b, "msg with args %d %s", 42, "42")        // want "compares: use assert\\.Greaterf"
		assert.GreaterOrEqual(t, a, b)                                   // want "compares: use assert\\.GreaterOrEqual"
		assert.GreaterOrEqualf(t, a, b, "msg with args %d %s", 42, "42") // want "compares: use assert\\.GreaterOrEqualf"
		assert.Less(t, a, b)                                             // want "compares: use assert\\.Less"
		assert.Lessf(t, a, b, "msg with args %d %s", 42, "42")           // want "compares: use assert\\.Lessf"
		assert.LessOrEqual(t, a, b)                                      // want "compares: use assert\\.LessOrEqual"
		assert.LessOrEqualf(t, a, b, "msg with args %d %s", 42, "42")    // want "compares: use assert\\.LessOrEqualf"
		assert.NotEqual(t, a, b)                                         // want "compares: use assert\\.NotEqual"
		assert.NotEqualf(t, a, b, "msg with args %d %s", 42, "42")       // want "compares: use assert\\.NotEqualf"
		assert.Equal(t, a, b)                                            // want "compares: use assert\\.Equal"
		assert.Equalf(t, a, b, "msg with args %d %s", 42, "42")          // want "compares: use assert\\.Equalf"
		assert.LessOrEqual(t, a, b)                                      // want "compares: use assert\\.LessOrEqual"
		assert.LessOrEqualf(t, a, b, "msg with args %d %s", 42, "42")    // want "compares: use assert\\.LessOrEqualf"
		assert.Less(t, a, b)                                             // want "compares: use assert\\.Less"
		assert.Lessf(t, a, b, "msg with args %d %s", 42, "42")           // want "compares: use assert\\.Lessf"
		assert.GreaterOrEqual(t, a, b)                                   // want "compares: use assert\\.GreaterOrEqual"
		assert.GreaterOrEqualf(t, a, b, "msg with args %d %s", 42, "42") // want "compares: use assert\\.GreaterOrEqualf"
		assert.Greater(t, a, b)                                          // want "compares: use assert\\.Greater"
		assert.Greaterf(t, a, b, "msg with args %d %s", 42, "42")        // want "compares: use assert\\.Greaterf"
		assert.Equal(t, a, b)                                            // want "compares: use assert\\.Equal"
		assert.Equalf(t, a, b, "msg with args %d %s", 42, "42")          // want "compares: use assert\\.Equalf"
		assert.NotEqual(t, a, b)                                         // want "compares: use assert\\.NotEqual"
		assert.NotEqualf(t, a, b, "msg with args %d %s", 42, "42")       // want "compares: use assert\\.NotEqualf"
		assert.Greater(t, a, b)                                          // want "compares: use assert\\.Greater"
		assert.Greaterf(t, a, b, "msg with args %d %s", 42, "42")        // want "compares: use assert\\.Greaterf"
		assert.GreaterOrEqual(t, a, b)                                   // want "compares: use assert\\.GreaterOrEqual"
		assert.GreaterOrEqualf(t, a, b, "msg with args %d %s", 42, "42") // want "compares: use assert\\.GreaterOrEqualf"
		assert.Less(t, a, b)                                             // want "compares: use assert\\.Less"
		assert.Lessf(t, a, b, "msg with args %d %s", 42, "42")           // want "compares: use assert\\.Lessf"
		assert.LessOrEqual(t, a, b)                                      // want "compares: use assert\\.LessOrEqual"
		assert.LessOrEqualf(t, a, b, "msg with args %d %s", 42, "42")    // want "compares: use assert\\.LessOrEqualf"
		assert.NotEqual(t, a, b)                                         // want "compares: use assert\\.NotEqual"
		assert.NotEqualf(t, a, b, "msg with args %d %s", 42, "42")       // want "compares: use assert\\.NotEqualf"
		assert.Equal(t, a, b)                                            // want "compares: use assert\\.Equal"
		assert.Equalf(t, a, b, "msg with args %d %s", 42, "42")          // want "compares: use assert\\.Equalf"
		assert.LessOrEqual(t, a, b)                                      // want "compares: use assert\\.LessOrEqual"
		assert.LessOrEqualf(t, a, b, "msg with args %d %s", 42, "42")    // want "compares: use assert\\.LessOrEqualf"
		assert.Less(t, a, b)                                             // want "compares: use assert\\.Less"
		assert.Lessf(t, a, b, "msg with args %d %s", 42, "42")           // want "compares: use assert\\.Lessf"
		assert.GreaterOrEqual(t, a, b)                                   // want "compares: use assert\\.GreaterOrEqual"
		assert.GreaterOrEqualf(t, a, b, "msg with args %d %s", 42, "42") // want "compares: use assert\\.GreaterOrEqualf"
		assert.Greater(t, a, b)                                          // want "compares: use assert\\.Greater"
		assert.Greaterf(t, a, b, "msg with args %d %s", 42, "42")        // want "compares: use assert\\.Greaterf"
		assert.Same(t, ptrA, ptrB)                                       // want "compares: use assert\\.Same"
		assert.Samef(t, ptrA, ptrB, "msg with args %d %s", 42, "42")     // want "compares: use assert\\.Samef"
		assert.NotSame(t, ptrA, ptrB)                                    // want "compares: use assert\\.NotSame"
		assert.NotSamef(t, ptrA, ptrB, "msg with args %d %s", 42, "42")  // want "compares: use assert\\.NotSamef"
		assert.NotSame(t, ptrA, ptrB)                                    // want "compares: use assert\\.NotSame"
		assert.NotSamef(t, ptrA, ptrB, "msg with args %d %s", 42, "42")  // want "compares: use assert\\.NotSamef"
		assert.Same(t, ptrA, ptrB)                                       // want "compares: use assert\\.Same"
		assert.Samef(t, ptrA, ptrB, "msg with args %d %s", 42, "42")     // want "compares: use assert\\.Samef"
	}

	// Valid.
	{
		assert.Equal(t, a, b)
		assert.Equalf(t, a, b, "msg with args %d %s", 42, "42")
		assert.NotEqual(t, a, b)
		assert.NotEqualf(t, a, b, "msg with args %d %s", 42, "42")
		assert.Greater(t, a, b)
		assert.Greaterf(t, a, b, "msg with args %d %s", 42, "42")
		assert.GreaterOrEqual(t, a, b)
		assert.GreaterOrEqualf(t, a, b, "msg with args %d %s", 42, "42")
		assert.Less(t, a, b)
		assert.Lessf(t, a, b, "msg with args %d %s", 42, "42")
		assert.LessOrEqual(t, a, b)
		assert.LessOrEqualf(t, a, b, "msg with args %d %s", 42, "42")
		assert.Same(t, ptrA, ptrB)
		assert.Samef(t, ptrA, ptrB, "msg with args %d %s", 42, "42")
		assert.NotSame(t, ptrA, ptrB)
		assert.NotSamef(t, ptrA, ptrB, "msg with args %d %s", 42, "42")
	}

	// Ignored.
	{
		assert.True(t, c && d)
		assert.Truef(t, c && d, "msg with args %d %s", 42, "42")
		assert.False(t, d && c)
		assert.Falsef(t, d && c, "msg with args %d %s", 42, "42")
		assert.True(t, c || d)
		assert.Truef(t, c || d, "msg with args %d %s", 42, "42")
		assert.False(t, d || c)
		assert.Falsef(t, d || c, "msg with args %d %s", 42, "42")
	}
}
